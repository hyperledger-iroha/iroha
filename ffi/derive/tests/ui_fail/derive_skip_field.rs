use std::mem::MaybeUninit;

use getset::{Getters, Setters};
use iroha_ffi::{ffi_export, FfiConvert, FfiType};

/// FfiStruct
#[ffi_export]
#[derive(Clone, Setters, Getters, FfiType)]
#[getset(get = "pub")]
pub struct FfiStruct {
    /// a
    #[getset(set = "pub")]
    a: i32,
    /// b
    #[getset(skip)]
    b: u32,
}

fn main() {
    let s = FfiStruct { a: 42, b: 32 };

    let mut a = MaybeUninit::<*const i32>::uninit();
    let mut b = MaybeUninit::<*const u32>::uninit();

    unsafe {
        FfiStruct__a(FfiConvert::into_ffi(&s, &mut ()), a.as_mut_ptr());
        let a: &i32 = FfiConvert::try_from_ffi(a.assume_init(), &mut ()).unwrap();
        FfiStruct__set_a(
            FfiConvert::into_ffi(&mut s, &mut ()),
            FfiConvert::into_ffi(*a, &mut ()),
        );

        FfiStruct__b(FfiConvert::into_ffi(&s, &mut ()), b.as_mut_ptr());
        let b: &u32 = FfiConvert::try_from_ffi(b.assume_init(), &mut ()).unwrap();
        FfiStruct__set_b(
            FfiConvert::into_ffi(&mut s, &mut ()),
            FfiConvert::into_ffi(*b, &mut ()),
        );
    }
}
