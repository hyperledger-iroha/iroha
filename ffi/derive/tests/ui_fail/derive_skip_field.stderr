error[E0425]: cannot find function, tuple struct or tuple variant `FfiStruct__b` in this scope
  --> tests/ui_fail/derive_skip_field.rs:33:9
   |
7  | #[ffi_export]
   | ------------- similarly named function `FfiStruct__a` defined here
...
33 |         FfiStruct__b(FfiConvert::into_ffi(&s, &mut ()), b.as_mut_ptr());
   |         ^^^^^^^^^^^^ help: a function with a similar name exists: `FfiStruct__a`

error[E0425]: cannot find function, tuple struct or tuple variant `FfiStruct__set_b` in this scope
  --> tests/ui_fail/derive_skip_field.rs:35:9
   |
7  | #[ffi_export]
   | ------------- similarly named function `FfiStruct__set_a` defined here
...
35 |         FfiStruct__set_b(
   |         ^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `FfiStruct__set_a`
